transcom/mymove
# INT-B-20078: Office Queue Pill Button Filters

## [B-20077](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A950885)
## [B-20078](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A950886)
## [E-05405](https://www13.v1host.com/USTRANSCOM38/Epic.mvc/Summary?oidToken=Epic:934940)

## Summary

**INTEGRATION**

This PR is the full implementation of the office queue filter pill buttons [EPIC](https://www13.v1host.com/USTRANSCOM38/Epic.mvc/Summary?oidToken=Epic:934940).  This PR will allow users a quick and easy way to remove search filters applied for all office queue views while saving search filters per queue view and session.  It contains both [B-20077(UI)](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A950885) and [B-20078(session cache)](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A950886).  


Work was done to ensure the fill button filters are compatible with Cory's [new multi-select type ahead column filter column filter](https://github.com/transcom/mymove/pull/13127) and [Alex's HQ Role search feature](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A949829).


## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Login as new service counseling, TIO, TOO, HQ users
2. For each account, ensure pill button filters  work appropriately and accurately for all queue view. This PR does not address search views. Only queue views with column filter headers!
3. Verify pill buttons are rendered correctly for all uses involving remove/add use case for ALL column headers. Ensure input text boxes, dropdown, multi-select, type ahead multi-select controls.
4. Verify pill buttons are cached for each specific queue view. Toggle between different views and verify filters  are maintained. This applies to page size, page and sort.
5. Verify filter pill buttons display correct labels. Single value filter columns shall only display the label name while multi-select filters shall display  in the following format: ${type}(${filter_value}). ....ie Status (New Move x) , Status(Approved x) ...etc..
6. Verify "All x" filter button is only displayed when there is more than one filters applied. 
7. Logout. Verify cache values are deleted when user logs in again. Filters are only cached for  active session.
8. Allow application to timeout/idle. Verify cache values are deleted when user logs in again. Filters are only cached for  active session.

 More than one filter.
![Screenshot 2024-07-02 at 3 50 52 PM](https://github.com/transcom/mymove/assets/148127004/f879d971-9681-424b-b8ef-49d727fd7a3f)

Single filter:
![Screenshot 2024-07-02 at 3 52 16 PM](https://github.com/transcom/mymove/assets/148127004/8f1f5ded-de8e-4e19-bca1-24a5f61b7181)

One Muti-select filter
![Screenshot 2024-07-02 at 3 53 31 PM](https://github.com/transcom/mymove/assets/148127004/54ddbb1a-824a-4088-b78d-3b08318c4db4)

More than one Muti-select filter:
![Screenshot 2024-07-02 at 3 54 21 PM](https://github.com/transcom/mymove/assets/148127004/15b68174-b4bb-436b-9213-aa346b43b478)

### Frontend
